### PR TITLE
Add tool to predict kernel boot measurements

### PR DESCRIPTION
**Pull request by conradgrobler**
When we start a new VM we pass the kernel in a bzImage format, but the VMM splits it into 2 parts before passing it to the Stage 0 firmware. So the measurements by stage 0 are of these two derived parts.

This tool applies the same changes to that we can predict what both measurements would be for a specific bzImage kernel.

Example local run:

Stage 0 logs when running Oak Containers Hello World:

```
stage0 DEBUG: Kernel image digest: sha2-256:3786bf59635b0ce0c5e2bd18f0cef28cc243ddeb13195531c509b8f9e0f1d538
stage0 DEBUG: Kernel setup data digest: sha2-256:b42f3c704ec17c87c17560ed9c23a44a44507e19065f81e16c3379e67417fd34
```

Running the prediction tool against the same Linux kernel:

```
Kernel Image Measurement: sha2-256:3786bf59635b0ce0c5e2bd18f0cef28cc243ddeb13195531c509b8f9e0f1d538
Kernel Setup Data Measurement: sha2-256:b42f3c704ec17c87c17560ed9c23a44a44507e19065f81e16c3379e67417fd34
```

### CODE REVIEW

File: oak_kernel_measurement/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Kernel Measurement Prediction Tool
+
+When booting a VM the kernel and its associated setup data is measured by the
+Stage 0 firmware. The VMM accepts a single bzImage format kernel file. It then
```
Comment (thmsbinder) [inline oak_kernel_measurement/README.md:4]: VMM = Virtual Machine Manager (just looked it up). Probably fine to leave if it is commonplace.

File: oak_kernel_measurement/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+Stage 0 measures these split, modified components rather than the original
+bzImage kernel. This tool can be used to predict the Stage 0 measurements of
+these components from a bzImage kernel.
+
+The tool can be run using:
+
+```bash
+cargo run --package=oak_kernel_measurement -- \
+    --kernel=oak_containers_kernel/target/bzImage
```
Comment (thmsbinder) [inline oak_kernel_measurement/README.md:17]: Cool, thanks!

File: oak_kernel_measurement/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        let kernel_image = bz_image[setup_size..].to_vec();
+        let mut setup_data = bz_image[..setup_size].to_vec();
+        // The VMM will modify some fields in the setup data header.
+        // The loader type will be set to `QEMU`.
+        setup_data[0x210] = 0xB0;
+        // The load flags will be updated to enable heap usage.
+        setup_data[0x211] |= 0x80;
+        // Set the default command_line location.
+        let default_cmd_line = 0x20000;
+        *u32::mut_from(&mut setup_data[0x228..0x22C]).expect("invalid slice for cmd_line") =
```
Comment (thmsbinder) [inline oak_kernel_measurement/src/main.rs:94]: Are these addresses always the same? Like, the same for all Linux kernels?

**Review by thmsbinder — APPROVED** [review]

File: oak_kernel_measurement/src/main.rs (hunk 3 above)
Comment (conradgrobler) [reply oak_kernel_measurement/src/main.rs:94]: Yes, they are specified in the Linux boot protocol (https://www.kernel.org/doc/html/v6.7/arch/x86/boot.html). Newer versions might add new fields, but the offsets for existing fields don't change.

**Pull request merged**